src/WebUI/Common/KubeSummary.tsx

```diff
@@ -273,11 +273,65 @@ export class KubeSummary extends BaseComponent<IKubeSummaryProps, IKubernetesCon
-            showSelectedItem: selectionStoreState.showSelectedItem,
-            selectedItem: selectionStoreState.selectedItem,
-            selectedItemType: selectionStoreState.selectedItemType
-        });
+
+        // If the selection only has id and doesn't have the item, we would need to fetch the item
+        if (!!selectionStoreState.itemUID && !selectionStoreState.selectedItem) {
+            let invokedFunc: (() => Promise<V1DaemonSetList | V1ReplicaSetList | V1StatefulSetList>) | undefined = undefined;
+            let item: V1DaemonSet | V1ReplicaSet | V1StatefulSet | undefined = undefined;
+            switch (selectionStoreState.selectedItemType) {
+                case SelectedItemKeys.ReplicaSetKey:
+                    const replicaSets = this._workloadsStore.getState().replicaSetList;
+                    if (replicaSets && replicaSets.items && replicaSets.items.length) {
+                        item = replicaSets.items.filter(r => r.metadata.uid === selectionStoreState.itemUID)[0];
+                    }
+                    invokedFunc = () => KubeSummary.getKubeService().getReplicaSets();
+                    break;
+                case SelectedItemKeys.StatefulSetKey:
+                    const statefulSets = this._workloadsStore.getState().statefulSetList;
+                    if (statefulSets && statefulSets.items && statefulSets.items.length) {
+                        item = statefulSets.items.filter(s => s.metadata.uid === selectionStoreState.itemUID)[0];
+                    }
+                    invokedFunc = () => KubeSummary.getKubeService().getStatefulSets();
+                    break;
+                case SelectedItemKeys.DaemonSetKey:
+                    const daemonSets = this._workloadsStore.getState().daemonSetList;
+                    if (daemonSets && daemonSets.items && daemonSets.items.length) {
+                        item = daemonSets.items.filter(d => d.metadata.uid === selectionStoreState.itemUID)[0];
+                    }
+                    invokedFunc = () => KubeSummary.getKubeService().getDaemonSets();
+                    break;
+            }
+
+            if (item) {
+                this.setState({
+                    showSelectedItem: selectionStoreState.showSelectedItem,
+                    selectedItem: item,
+                    selectedItemType: selectionStoreState.selectedItemType
+                });
+            }
+            else if (invokedFunc) {
+                invokedFunc().then(list => {
+                    if (list && list.items && list.items.length) {
+                        item = (list.items as { metadata: V1ObjectMeta }[]).filter((i: { metadata: V1ObjectMeta }) => i.metadata.uid === selectionStoreState.itemUID)[0] as V1DaemonSet | V1ReplicaSet | V1StatefulSet;
+                    }
+
+                    if (item) {
+                        this.setState({
+                            showSelectedItem: selectionStoreState.showSelectedItem,
```

Comment:
showSelectedItem: selectionStoreState.showSelectedItem,
                            selectedItem: item,
                            selectedItemType: selectionStoreState.selectedItemType
                            
can we have a anonymous function taking item as value and setting the state?